proposals/0173-Read-Generic-Network-Signal-data.md

```diff
@@ -16,277 +16,306 @@ To enable OEM proprietary mobile apps to read raw CAN/Network data by specifying
+	<description>Reflects the status of the ambient light sensor.</description>
+	<element name="NIGHT" />
+	<element name="TWILIGHT_1" />
+	<element name="TWILIGHT_2" />
+	<element name="TWILIGHT_3" />
+	<element name="TWILIGHT_4" />
+	<element name="DAY" />
+	<element name="UNKNOWN" internal_name="ALS_UNKNOWN" />
+	<element name="INVALID" />
+</enum> 
 
-* For GetVehicleData response/OnVehicleData, add additional response structure:
-  * networkData - It has a paramId and value returned from vehicle. Value returned will be a Hex string.
-  
 ```
-<function name="GetVehicleData" functionID="GetVehicleDataID" messagetype="response">
 
-    <param name="success" type="Boolean" platform="documentation">
-		<description> true, if successful; false, if failed </description>
-	</param>
-
-    <param name="resultCode" type="Result" platform="documentation">
-      <description>See Result</description>
-      <element name="SUCCESS"/>
-      <element name="INVALID_DATA"/>
-      <element name="OUT_OF_MEMORY"/>
-      <element name="TOO_MANY_PENDING_REQUESTS"/>
-      <element name="APPLICATION_NOT_REGISTERED"/>
-      <element name="GENERIC_ERROR"/>
-      <element name="REJECTED"/>
-      <element name="VEHICLE_DATA_NOT_AVAILABLE"/>
-      <element name="USER_DISALLOWED"/>
-    </param>
+* As you see, there are 3 sub items, “ambientLightSensorStatus”, “highBeamsOn” and “lowBeamsOn”. Each of these individual data types and references
+* You many notice that “ambientLightSensorStatus” has data type as “AmbientLightStatus”. This enumeration is read from existing Mobile API.
+  * This can be extrapolated to all existing data types. For new data types enumerations, we’d still either need to add to Mobile API or create a new structure for vehicle data enumerations.
+
+### Next example touches on a *new* OEM specific vehicle data item, _engineState_
+
+Sample value for map:
+```json
+{
+	"id":"engineState",
+	"dataType":"Struct",
+	"reference":"OEM_REF_STRUCT",
+	"engineCoolantTemp" : {
+		"dataType":"Integer",
+		"reference":"OEM_REF_ENG_COOL_TEMP",
+		"array":false,
+		"mandatory":false
+	},
+	"engineO2Sat":{
+		"dataType":"Float",
+		"reference":"OEM_REF_ENG_O2_SAT",				
+		"mandatory":false
+	},
+	"engineState":{
+		"dataType":"String",
+		"reference":"OEM_REF_ENG_STATE"
+	},
+	"engineServiceRequired":{
+		"dataType":"Boolean",
+		"reference":"OEM_REF_ENG_SER_REQ",
+		"array":false,
+		"mandatory":false
+	}
+}
+```
+Sample response from module:
+```json
+{
+	"engineState": {
+		"engineCoolantTemp":140,
+		"engineO2Sat":95.5,
+		"engineState":"NORMAL",
+		"engineServiceRequired":false
+	}
+}
+```
+#### engineState would ideally be an enum if it was a Standardized data type. But since this it OEM example of vehicle data on the fly, we can use String data type.
+
+* _engineState_ would need to be added to both SDL core and proxy schema. Once new schema is downloaded by both core and proxy, an app may request _engineState_ vehicle data item.
```

Comment:
Change to: 
_engineState_ would need to be added to both SDL core and proxy schema. Once **the** new schema is downloaded by both core and proxy, an app may request _engineState_ vehicle data item.

Reply:
Update this please

Reply:
Update to:
Once **the** new schema is downloaded by both core and proxy, an app may request engineState vehicle data item.